src/buildCommon.js

```diff
@@ -632,7 +639,7 @@ const retrieveTextFontName = function(
             baseFontName = "Typewriter";
             break;
         default:
-            throw new Error(`Invalid font provided: ${fontFamily}`);
+            baseFontName = fontFamily; // use fonts added by a plugin
```

Comment:
It would be nice if we could check when we get an invalid font.  Right now there's no list of valid fontFamilies.  Ideally we should have one and setFontMetrics should update that list with the font that's being added.  Also, this makes me wonder if we should change `__setFontMetrics` to be `__defineFont` and have it take something a little more structured, e.g.:
```
__defineFont(
  baseFontName, 
  fontFamily, 
  metrics: { regular: {...}, bold: {...}, italic: {...}, "italic-bold": {...} },
);
```
That's a rather big change.  Would you be able to tackle this in a follow up PR?

Reply:
Yes, good idea.  
What is the difference between baseFontName and fontFamily here?